### PR TITLE
add metric for monitoring email notification failure

### PR DESCRIPTION
**Pull request by houqp**
This should help catch unexpected email notification failures. We need to monitor the monitoring system :)

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by jstern** [comment]
Seems useful :)

Wondering if you'd want anything similar on failing to send an email when email_on_failure is true? Or maybe even stats for errors running callbacks in general?

**Comment by houqp** [comment]
yeah, it would be useful to have metrics for all notification errors. that said, the ones you mentioned are mostly from executing task instances, not from scheduler itself, so I think it's better to leave those changes to a separate PR :)

File: docs/metrics.rst [hunk 1]
```diff
@@ -85,6 +85,7 @@ Name                                    Description
 ``scheduler.tasks.killed_externally``   Number of tasks killed externally
 ``scheduler.tasks.running``             Number of tasks running in executor
 ``scheduler.tasks.starving``            Number of tasks that cannot be scheduled because of no open slot in pool
+``sla_email_notification_failure``      Number failed SLA miss email notification attempts
```
Comment (kaxil) [inline docs/metrics.rst]:
```suggestion
``sla_email_notification_failure``      Number of failed SLA miss email notification attempts
```

**Review by kaxil — APPROVED** [review]
LGTM except 1 minor change suggested

**Comment by houqp** [comment]
@kaxil good catch, updated :)

**Pull request merged**